src/main/java/picard/sam/RevertSam.java

```diff
@@ -580,30 +577,27 @@ private SAMFileHeader createOutHeader(
         return outHeader;
     }
 
-    private Map<SAMReadGroupRecord, FastqQualityFormat> createReadGroupFormatMap(
-            final SAMFileHeader inHeader,
-            final File referenceSequence,
-            final ValidationStringency validationStringency,
-            final File input,
-            final boolean restoreOriginalQualities) {
+    private Map<SAMReadGroupRecord,FastqQualityFormat> createReadGroupFormatMap(RevertSamSorter sorter, boolean restoreOriginalQualities) {
 
         final Map<SAMReadGroupRecord, FastqQualityFormat> readGroupToFormat = new HashMap<>();
 
-        // Figure out the quality score encoding scheme for each read group.
-        for (final SAMReadGroupRecord rg : inHeader.getReadGroups()) {
-            final SamReader reader = SamReaderFactory.makeDefault().referenceSequence(referenceSequence).validationStringency(validationStringency).open(input);
-            final SamRecordFilter filter = new SamRecordFilter() {
-                public boolean filterOut(final SAMRecord rec) {
-                    return !rec.getReadGroup().getId().equals(rg.getId());
-                }
+        List<PeekableIterator<SAMRecord>> iterators = sorter.iterators();
+        for (PeekableIterator<SAMRecord> iterator : iterators) {
+            SAMReadGroupRecord readGroup = iterator.peek().getReadGroup();
```

Comment:
`iterators`, `iterator`, and `readGroup` can/should be marked `final`.

Reply:
Agreed and added, thanks!